### PR TITLE
Framework: Removed sites-list dependency from lib/posts/actions.js take 2

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This is a take 2 of https://github.com/Automattic/wp-calypso/pull/16017, unfortunately, that version had a bug that affected pages (not posts) and while doing the tests on staging I found it and had to revert. The PR removes sites-list from posts/actions. Also, makes some actions receive site object and makes all the required components pass it.
This is required to allow the removal of sites-list from other parts of lib/posts.

The only file change since the approval is update-post-status/index.js.

To test:
Create a new post, publish it, revert to draft, move to trash, restore it, delete permanently, change it, autosave it etc...
Do the same for pages.
Everything that changes posts and/or pages should be exhaustively tested to make sure the functionality was not broken.
Execute the automated tests in lib posts:
npm run test-client client/lib/posts

### CODE REVIEW

**Comment by Copons** [comment]
Tested and it works on both posts and pages, but I've compared the changes between this and #16017: they look exactly the same. 🤔

**Comment by jorgefilipecosta** [comment]
Hi @Copons thank you for reviewing and testing this PR. The difference when compared to the other version is minor (but with big impact) on file client/components/update-post-status/index.js on line 
 56 instead of "site: getSite( state, get( props, 'post.site_ID' ) )," in this one we have "site: getSite( state, get( props, [ props.page ? 'page' : 'post', 'site_ID' ] ) ),".

**Comment by Copons** [comment]
Oh shoot, totally missed that!

Then this LGTM again - and hopefully for good this time!
I'm super sorry for not catching the issue that led to the revert on the other PR. 🙁

**Comment by jorgefilipecosta** [comment]
Hi @Copons, the problem was never in production I reverted before the deploy to production and found it while testing on stage so it did not impact our users. Thank you a lot for your tests, this one was easy to miss I focused on testing posts and unfortunately missed the page case.

File: client/post-editor/controller.js [hunk 1]
```diff
@@ -210,6 +210,7 @@ export default {
 		const postToCopyId = context.query.copy;
 
 		function startEditing( siteId ) {
+			const site = getSelectedSite( context.store.getState() );
```
Comment (jsnajdr) [inline client/post-editor/controller.js]: If the `siteId` is explicitly received as parameter, wouldn't it make more sense to call `getSite( state, siteId)` here? I know that @Copons gave you exactly the opposite advice (:smile:) and that in practice it doesn't really matter (the site will be always the selected one anyway). But it looks quite confusing to have a `siteId` parameter and then ignore it.

File: client/lib/posts/actions.js [hunk 2]
```diff
@@ -100,19 +98,20 @@ PostActions = {
...
 		options = options || {};
 
 		args = {
 			type: 'DRAFT_NEW_POST',
-			siteId: siteId,
+			siteId: site.ID,
 			postType: options.type || 'post',
 			title: options.title,
 			content: options.content,
+			site,
```
Comment (jsnajdr) [inline client/lib/posts/actions.js]: Does the `site` really need to be passed as property of the `DRAFT_NEW_POST` action? It doesn't seem to be used in any of the stores that handle it. It doesn't do any harm if it's there, but is most likely redundant.

File: client/lib/posts/actions.js [hunk 3]
```diff
@@ -121,34 +120,35 @@ PostActions = {
...
 
-		postHandle = wpcom.site( siteId ).post( postId );
+		postHandle = wpcom.site( site.ID ).post( postId );
 
 		postHandle.get( { context: 'edit', meta: 'autosave' }, function( error, data ) {
 			Dispatcher.handleServerAction( {
 				type: 'RECEIVE_POST_TO_EDIT',
 				error: error,
 				post: data,
+				site,
```
Comment (jsnajdr) [inline client/lib/posts/actions.js]: ditto

File: client/lib/posts/actions.js [hunk 4]
```diff
@@ -205,13 +202,17 @@ PostActions = {
 						type: 'RECEIVE_POST_AUTOSAVE',
 						error: error,
 						autosave: data,
+						site,
```
Comment (jsnajdr) [inline client/lib/posts/actions.js]: ditto

File: client/lib/posts/actions.js [hunk 5]
```diff
@@ -392,6 +394,7 @@ PostActions = {
 				isNew: isNew,
 				original: original,
 				post: data,
+				site,
```
Comment (jsnajdr) [inline client/lib/posts/actions.js]: ditto

File: client/lib/posts/actions.js [hunk 6]
```diff
@@ -538,6 +545,7 @@ PostActions = {
 			error: error,
 			original: original,
 			post: data,
+			site,
```
Comment (jsnajdr) [inline client/lib/posts/actions.js]: ditto

File: client/lib/posts/actions.js (hunk 2 above)
Comment (jsnajdr) [reply client/lib/posts/actions.js]: I get it after seeing #16019 -- the `site` param is indeed needed. I'd only remove the redundant `siteId` param from the `DRAFT_NEW_POST` action.

**Pull request merged**